### DISCUSSION TITLE
Native look and feel on KDE Plasma

### DISCUSSION THREAD

**Discussion by CarterLi** [ghostty-org/ghostty#3663] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/805f0ba0-ae2f-4941-a6bf-b7213f636ef2)

The title bar of ghostty seems too large on KDE, comparing to other native KDE applications.

In addition, the cursor is very large. I don't know if it's because I have DPI scaling enabled.

**Comment by 00-kat** [comment]
Title bar: Ghostty is a libadwaita app, and there is currently no Qt apprt :(  but try modifying [gtk-titlebar](https://ghostty.org/docs/config/reference#gtk-titlebar), [window-decoration](https://ghostty.org/docs/config/reference#window-decoration) and/or [gtk-adwaita](https://ghostty.org/docs/config/reference#gtk-adwaita).

Cursor: seems to be an issue with GTK and Wayland but no one seems to know.  Relevant threads on the [Ghostty Discord server](https://discord.gg/ghostty): [Wayland cursor scaling issues](https://discord.com/channels/1005603569187160125/1312128716410982451), [Large cursor and visual artifacts under wayland, on arch KDE](https://discord.com/channels/1005603569187160125/1308878017409581086), [Fractional Scaling on Linux](https://discord.com/channels/1005603569187160125/1322151734747070475).

**Reply by SuperDuperDeou** [reply]
The reason why GTK cursors are bad on Wayland is that GTK doesn't support server-side cursor, and instead of setting the client cursor via the `$XCURSOR_SIZE` and `$XCURSOR_THEME` like anything sane, it insists on using dconf.

And then there's fractional scaling, GTK's fractional scaling might as well not be there.

**Reply by 00-kat** [reply]
For me, without changing the cursor with `gsettings` (i.e.: dconf), all GTK apps such as Firefox also used an incorrect cursor.  Many people are reporting this to be an issue only in Ghostty, though.

**Reply by CarterLi** [reply]
Firefox works for me. Even GNOME terminal works perfectly.

**Reply by 00-kat** [reply]
Somebody mentioned in another thread that it might be because this only affects GTK4 apps and not GTK3 ones.

Someone else also linked this forum thread: https://bbs.archlinux.org/viewtopic.php?id=299624 (TL;DR: appears to be a GTK4 bug with the Breeze cursor theme (i.e.: plasma default)).

**Comment by tobyrg** [comment]
It would be great if Ghostty supported SSD (Server side decorations) this would solve this as the compositor would provide the title bar. This would make it fit in better in both KDE Plasma and other desktop environments such as COSMIC.

**Comment by ESUAdmin** [comment]
将军 you can `env GDK_SCALE= ghostty` and double window-{width,height} values in config

**Reply by phush0** [reply]
Everything using scale is based on default cursor of 24 multiplied by scalling, my is 1.75 and cursor is huge. Eta fix for this is next GTK release in March. You can try set other cursor theme some are ok some are not.

**Comment by wlh320** [comment]
You can set `gtk-titlebar = false` to use kwin's titlebar instead of GTK's. It works in my KDE environment.

**Reply by CarterLi** [reply]
It does work. However, the useful buttons `View Open Tabs` and `Main Menu` disappear. In addition, the style of tabs bar seems off.

**Reply by wlh320** [reply]
I think the buttons are with the GTK titlebar. Currently it cannot have both KDE titlebar and these buttons. As for the tabs, trying different `window-theme` settings may help. I also encountered this problem and I don't know what the reason was, but it later returned to normal on its own.

**Reply by 00-kat** [reply]
#2642 and #3434 are probably relevant.

**Reply by VinceCris** [reply]
> It does work. However, the useful buttons `View Open Tabs` and `Main Menu` disappear. In addition, the style of tabs bar seems off.

I used `gtk-adwaita = false` which admittedly still removes the `View Open Tabs` option, but only that. Additionally, I downloaded and installed the GTK/GNOME Theme using the KDE System Settings  -> Global theme -> Application Style

Also attempted `keybind = ctrl+m=toggle_tab_overview` but that's currently supported only when `gtk-adwaita` is set to `true` unfortunately.

**Comment by pluiedev** [comment, marked as answer]
SSD support for Wayland has been added in 1.1: https://ghostty.org/docs/install/release-notes/1-1-0#server-side-decorations-(ssd)-on-linux. X11 support is scheduled for 1.1.1: see PR #5533
